### ISSUE TITLE
Mail only sends UUID instead of Username

### ISSUE THREAD

**Issue by DalePayman** [nextcloud/server#3323]
### Steps to reproduce
1. Upgrade from v10 to 11

### Expected behaviour
The Language is german - so the system should send german mails with the Username instead of UUID

### Actual behaviour
The language is english - i think it uses this template (core/templates/mail.php) and instead of username it writes users UUID.
See: https://help.nextcloud.com/uploads/default/original/2X/a/a9efde92d6864f20c48238b1578f06018183e4e9.PNG

### Server configuration
**Operating system**:
CentOS Linux release 7.2.1511 (Core)

**Web server:**
Server version: Apache/2.4.6 (CentOS)
Server built:   Jul 18 2016 15:30:14

**Database:**
5.5.50-MariaDB

**PHP version:**
PHP 7.0.10

**Nextcloud version:** (see Nextcloud admin page)
11.0.1 (stable)

**Updated from an older Nextcloud/ownCloud or fresh install:**
Updated from 10

**Where did you install Nextcloud from:**
Nextcloud download page.

**Signing status:**

```
No errors have been found.
```
</details>

**List of activated apps:**

```
Enabled:
  - activity: 2.4.1
  - comments: 1.1.0
  - dav: 1.1.1
  - encryption: 1.4.1
  - federatedfilesharing: 1.1.1
  - files: 1.6.1
  - files_pdfviewer: 1.0.1
  - files_sharing: 1.1.1
  - files_texteditor: 2.2
  - files_trashbin: 1.1.0
  - files_versions: 1.4.0
  - files_videoplayer: 1.0.0
  - firstrunwizard: 2.0
  - gallery: 16.0.0
  - logreader: 2.0.0
  - lookup_server_connector: 1.0.0
  - nextcloud_announcements: 1.0
  - notifications: 1.0.1
  - password_policy: 1.1.0
  - provisioning_api: 1.1.0
  - serverinfo: 1.1.1
  - sharebymail: 1.0.1
  - survey_client: 0.1.5
  - systemtags: 1.1.3
  - theming: 1.1.1
  - twofactor_backupcodes: 1.0.0
  - updatenotification: 1.1.1
  - user_ldap: 1.1.1
  - workflowengine: 1.1.1
Disabled:
  - admin_audit
  - external
  - federation
  - files_accesscontrol
  - files_automatedtagging
  - files_external
  - files_retention
  - templateeditor
  - user_external
  - user_saml
```
</details>

**The content of config/config.php:**

```
{
    "system": {
        "instanceid": "XXXX",
        "logfile": "\/var\/log\/nextcloud.log",
        "loglevel": 2,
        "memcache.local": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "localhost",
            "port": XXXXX
        },
        "passwordsalt": "XXXXX",
        "secret": "XXXXX",
        "trusted_domains": [
            "drive.XX.XX"
        ],
        "datadirectory": "\/opt\/XXXX\/files",
        "overwrite.cli.url": "https:\/\/XX.XX",
        "dbtype": "mysql",
        "version": "11.0.1.2",
        "dbname": "nextcloud",
        "dbhost": "localhost",
        "dbport": "",
        "dbtableprefix": "oc_",
        "dbuser": "XXXX",
        "dbpassword": "XXXX",
        "logtimezone": "UTC",
        "installed": true,
        "ldapIgnoreNamingRules": false,
        "ldapProviderFactory": "XXX",
        "mail_from_address": "no-reply",
        "mail_smtpmode": "smtp",
        "mail_domain": "XX.XX",
        "maintenance": false,
        "default_language": "de_DE",
        "htaccess.RewriteBase": "\/",
        "theme": "XXX_nextcloud_theme",
        "mail_smtphost": "XXXXX",
        "mail_smtpport": "25"
    }
}
```
</details>

**Are you using external storage, if yes which one:** local/smb/sftp/...
No

**Are you using encryption:** yes/no
Yes

**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...
LDAP

#### LDAP configuration (delete this part if not used)
<details>
<summary>LDAP config</summary>

```
+-------------------------------+--------------------------------------------------------------------------------------------------------------------------------+
| Configuration                 | s01                                                                                                                            |
+-------------------------------+--------------------------------------------------------------------------------------------------------------------------------+
| hasMemberOfFilterSupport      | 1                                                                                                                              |
| hasPagedResultSupport         |                                                                                                                                |
| homeFolderNamingRule          |                                                                                                                                |
| lastJpegPhotoLookup           | 0                                                                                                                              |
| ldapAgentName                 | cn=,ou=,dc=,dc= 			                                                                                         |
| ldapAgentPassword             | ***                                                                                                                            |
| ldapAttributesForGroupSearch  |                                                                                                                                |
| ldapAttributesForUserSearch   |                                                                                                                                |
| ldapBackupHost                |                                                                                                                                |
| ldapBackupPort                |                                                                                                                                |
| ldapBase                      |                                                                                                                                |
| ldapBaseGroups                |                                                                                                                                |
| ldapBaseUsers                 |                                                                                                                                |
| ldapCacheTTL                  | 600                                                                                                                            |
| ldapConfigurationActive       | 1                                                                                                                              |
| ldapDynamicGroupMemberURL     |                                                                                                                                |
| ldapEmailAttribute            | mail                                                                                                                           |
| ldapExperiencedAdmin          | 0                                                                                                                              |
| ldapExpertUUIDGroupAttr       |                                                                                                                                |
| ldapExpertUUIDUserAttr        |                                                                                                                                |
| ldapExpertUsernameAttr        |                                                                                                                                |
| ldapGroupDisplayName          | cn                                                                                                                             |
| ldapGroupFilter               | (|(cn=))                                                                                                                       |
| ldapGroupFilterGroups         |                                                                                                                                |
| ldapGroupFilterMode           | 0                                                                                                                              |
| ldapGroupFilterObjectclass    |                                                                                                                                |
| ldapGroupMemberAssocAttr      | member                                                                                                                         |
| ldapHost                      |                                                                                                                                |
| ldapIgnoreNamingRules         |                                                                                                                                |
| ldapLoginFilter               | (&(&(|(objectclass=person))(|(|(memberof=CN=,OU=,DC=,DC=)(primaryGroupID=))))(samaccountname=%uid))                            |
| ldapLoginFilterAttributes     |                                                                                                                                |
| ldapLoginFilterEmail          | 0                                                                                                                              |
| ldapLoginFilterMode           | 1                                                                                                                              |
| ldapLoginFilterUsername       | 1                                                                                                                              |
| ldapNestedGroups              | 0                                                                                                                              |
| ldapOverrideMainServer        |                                                                                                                                |
| ldapPagingSize                | 500                                                                                                                            |
| ldapPort                      | 389                                                                                                                            |
| ldapQuotaAttribute            |                                                                                                                                |
| ldapQuotaDefault              |                                                                                                                                |
| ldapTLS                       | 0                                                                                                                              |
| ldapUserDisplayName           | displayname                                                                                                                    |
| ldapUserDisplayName2          |                                                                                                                                |
| ldapUserFilter                | (&(|(objectclass=person))(|(|(memberof=CN=,OU=,DC=,DC=)(primaryGroupID=12680))))                                               |
| ldapUserFilterGroups          | Drive                                                                                                                          |
| ldapUserFilterMode            | 0                                                                                                                              |
| ldapUserFilterObjectclass     | person                                                                                                                         |
| ldapUuidGroupAttribute        | auto                                                                                                                           |
| ldapUuidUserAttribute         | auto                                                                                                                           |
| turnOffCertCheck              | 0                                                                                                                              |
| turnOnPasswordChange          | 0                                                                                                                              |
| useMemberOfToDetectMembership | 1                                                                                                                              |
+-------------------------------+--------------------------------------------------------------------------------------------------------------------------------+
```
</details>

### Client configuration
**Browser:**
Firefox & Chrome

**Operating system:**
Mac & CentOS

#### Web server error log
<details>
<summary>Web server error log</summary>
```
don't think this is necessary
```
</details>

#### Nextcloud log (data/nextcloud.log)
<details>
<summary>Nextcloud log</summary>
```
don't think this is necessary
```
</details>

#### Browser log
<details>
<summary>Browser log</summary>
```
don't think this is necessary
```
</details>

**Comment by MorrisJobke** [comment]
The username is fixed with #3180 (will be released with 11.0.2).

For the language maybe @nickvergessen knows more.

**Comment by DalePayman** [comment]
Thx!
